linkerd/app/inbound/src/policy/http.rs

```diff
@@ -215,6 +215,17 @@ impl<T, N> HttpPolicyService<T, N> {
                     client.ip = %self.connection.client.ip(),
                     "Request denied",
                 );
+                if route.authorizations.is_empty() {
+                    tracing::debug!("No authorizations defined",);
+                }
+                for authz in &*route.authorizations {
+                    tracing::debug!(
+                        authz.group = %authz.meta.group(),
+                        authz.kind = %authz.meta.kind(),
+                        authz.name = %authz.meta.name(),
+                        "Authorization did not apply",
+                    );
+                }
```

Comment:
maybe worth sticking all of this code behind a `if tracing::event_enabled!(Level::DEBUG)`, since the loop will still run if `DEBUG` is not enabled?
```suggestion
                if tracing::event_enabled!(Level::DEBUG) {
                    if route.authorizations.is_empty() {
                        tracing::debug!("No authorizations defined",);
                    }
                    for authz in &*route.authorizations {
                        tracing::debug!(
                            authz.group = %authz.meta.group(),
                            authz.kind = %authz.meta.kind(),
                            authz.name = %authz.meta.name(),
                            "Authorization did not apply",
                        );
                    }
                }
```